uefi/src/main.rs

```diff
@@ -472,25 +481,82 @@ fn init_logger(
 
     let mode = {
         let modes = gop.modes();
-        match (
-            config
-                .frame_buffer
-                .minimum_framebuffer_height
-                .map(|v| usize::try_from(v).unwrap()),
-            config
-                .frame_buffer
-                .minimum_framebuffer_width
-                .map(|v| usize::try_from(v).unwrap()),
-        ) {
-            (Some(height), Some(width)) => modes
-                .filter(|m| {
-                    let res = m.info().resolution();
-                    res.1 >= height && res.0 >= width
-                })
-                .last(),
-            (Some(height), None) => modes.filter(|m| m.info().resolution().1 >= height).last(),
-            (None, Some(width)) => modes.filter(|m| m.info().resolution().0 >= width).last(),
-            _ => None,
+
+        if let Some(hypervisor) = CpuId::new().get_hypervisor_info() {
+            if let Hypervisor::Xen = hypervisor.identify() {
+                // Use same rules as real hardware since Xen uses the whole screen
```

Comment:
I'm a bit puzzled about what this PR is doing.

I often use VMWare and QEMU/KVM for running guest VMs as I would a host, especially when enabling paravirtualization features for performance similar to host system. These guests may run windowed or fullscreen.

---

It's not too concerning for me as this is only applicable to a bootloader based on this project, but the motivation seems focused on QEMU windowed for development/testing purposes?

As a user, if I have a VM guest fullscreen on a display(s), I'd find this different implicit behaviour a bit confusing, especially since there's an exception made for Xen.

I don't think you find GRUB, rEFInd, or systemd-boot handling resolution used differently? I believe they have a config setting if you want to explicitly prefer a resolution or scaling/fit?

I'm not that familiar with the project, but is this just a default with a way to opt-out? Is there actual value in a physical + virtual framebuffer structs that this PR introduces for this distinction?

Or would it be better to match what other bootloader/managers do, and just provide a default with build (or runtime) config to have the scaling behaviour you prefer?

You could then more easily build for your VM testing, with a different config for production builds on real hardware (or VM guests). The detection to switch based on environment if desired beyond testing may be better served as an opt-in feature/config?

---

If you disagree, no worries 👍 

Maybe consider documenting the behaviour then so it's easier to troubleshoot when a dev/user encounters it and tries to understand why it scales differently on most hypervisors.